docs/best-practice/gcs-fault-tolerance-persistent-redis.md

```diff
@@ -0,0 +1,127 @@
+# Fault Tolerant GCS with persistent Redis
+
+Using Redis to back up the Global Control Store (GCS) with KubeRay provides
+fault tolerance in the event that the Ray Head is lost, allowing the new Ray
+Head to rebuild its state from reading Redis.
+
+However, if Redis also has data loss, the Ray Head state will be lost.
+
+You may want further protection in the event that your Redis cluster experiences
+partial or total failure. This guide documents how to configure and tune Redis
+for a highly available Ray Cluster with KubeRay.
+
+Tuning your ray cluster to be highly available can safeguard long-running jobs
+against unexpected failures or allow you to run on commodity
+hardware/pre-emptible machines.
+
+## Solution overview
+
+KubeRay supports using Redis to persist the GCS, which allows us to move the
+point of failure (for data loss) externally. We still have to configure Redis
+itself to be resilient to failures.
+
+Our solution will provision a [Persistent
+Volume](https://kubernetes.io/docs/concepts/storage/persistent-volumes/) backed
+by hardware storage, which Redis will use to write regular snapshots. If Redis
+(or its host node) is lost, the Redis deployment can be restored from the
+snapshot.
+
+While Redis supports clustering, KubeRay only supports standalone (single
+replica) Redis, so clustering is omitted.
+
+## Persistent storage
+
+Specialty storage volumes (like Google Cloud Storage FUSE or S3) are not
+recommended because append operations are not supported, which Redis uses to
+efficiently write its Append Only File (AOF) log.
```

Comment:
Instead of not recommending these cloud storage, can we say we recommend disabling AOF when backed by these cloud storages?